src/components/guideLineLayer.ts

```diff
@@ -0,0 +1,154 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Components {
+  export class GuideLineLayer<D> extends Component {
+    public static ORIENTATION_VERTICAL = "vertical";
+    public static ORIENTATION_HORIZONTAL = "horizontal";
+
+    private _orientation: string;
+    private _value: D;
+    private _scale: QuantitativeScale<D>;
+    private _pixelPosition: number;
+    private _scaleUpdateCallback = () => {
+      this._updatePixelPosition();
+      this.render();
+    };
+    private _line: d3.Selection<void>;
+
+    constructor(orientation: string) {
```

Comment:
Supposedly you can type the constructors like the below:

```typescript
constructor(orientation: "vertical");
constructor(orientation: "horizontal");
constructor(orientation: string);
```

Note the fact that the `string` version is still forced to appear.  Is this preferable?  Precedent set by `linePlot.interpolator()`